src/components/ble/AppleNotificationCenterClient.cpp

```diff
@@ -0,0 +1,514 @@
+    request[13] = ((messageSize >> 8) & 0xFF);
+
+    ble_gattc_write_flat(event->notify_rx.conn_handle, controlPointHandle, request, sizeof(request), OnControlPointWriteCallback, this);
+  } else if (event->notify_rx.attr_handle == dataSourceHandle || event->notify_rx.attr_handle == dataSourceDescriptorHandle) {
+    uint16_t titleSize;
+    uint16_t subTitleSize;
+    uint16_t messageSize;
+    uint32_t notificationUid;
+
+    os_mbuf_copydata(event->notify_rx.om, 1, 4, &notificationUid);
+    os_mbuf_copydata(event->notify_rx.om, 6, 2, &titleSize);
+    os_mbuf_copydata(event->notify_rx.om, 8 + titleSize + 1, 2, &subTitleSize);
+    os_mbuf_copydata(event->notify_rx.om, 8 + titleSize + 1 + 2 + subTitleSize + 1, 2, &messageSize);
+
+    AncsNotification ancsNotif;
+    ancsNotif.uuid = 0;
+
+    if (notifications.contains(notificationUid)) {
+      ancsNotif = notifications[notificationUid];
+    }
+
+    std::string decodedTitle = DecodeUtf8String(event->notify_rx.om, titleSize, 8);
+
+    std::string decodedSubTitle = DecodeUtf8String(event->notify_rx.om, subTitleSize, 8 + titleSize + 1 + 2);
+
+    std::string decodedMessage = DecodeUtf8String(event->notify_rx.om, messageSize, 8 + titleSize + 1 + 2 + subTitleSize + 1 + 2);
+
+    NRF_LOG_INFO("Decoded Title: %s", decodedTitle.c_str());
+    NRF_LOG_INFO("Decoded SubTitle: %s", decodedSubTitle.c_str());
+
+    bool incomingCall = ancsNotif.uuid != 0 && ancsNotif.category == static_cast<uint8_t>(Categories::IncomingCall);
+
+    if (!incomingCall) {
+      if (titleSize >= maxTitleSize) {
+        decodedTitle.resize(maxTitleSize - 3);
+        decodedTitle += "...";
+        if (!decodedSubTitle.empty()) {
+          decodedTitle += " - ";
+        } else {
+          decodedTitle += "-";
+        }
+      } else {
+        if (!decodedSubTitle.empty()) {
+          decodedTitle += " - ";
+        } else {
+          decodedTitle += "-";
+        }
+      }
+
+      if (subTitleSize > maxSubtitleSize) {
+        decodedSubTitle.resize(maxSubtitleSize - 3);
+        decodedSubTitle += "...";
+      }
+
+      // if (messageSize > maxMessageSize) {
+      //   decodedMessage.resize(maxMessageSize - 3);
+      //   decodedMessage += "...";
+      // }
+    }
+
+    titleSize = static_cast<uint16_t>(decodedTitle.size());
+    subTitleSize = static_cast<uint16_t>(decodedSubTitle.size());
+    messageSize = static_cast<uint16_t>(decodedMessage.size());
+
+    NotificationManager::Notification notif;
+    notif.ancsUid = notificationUid;
+    std::string notifStr;
+
+    if (incomingCall) {
+      notifStr += "Incoming Call:";
+      notifStr += decodedTitle;
+      notifStr += "\n";
+      notifStr += decodedSubTitle;
+    } else {
+      notifStr += decodedTitle;
+      if (!decodedSubTitle.empty()) {
+        notifStr += decodedSubTitle + ":";
+      }
+      notifStr += decodedMessage;
+    }
+
+    // Adjust notification if too long
+    if (notifStr.size() > 100) {
+      notifStr.resize(97);
```

Comment:
`NotificationManager` has an attribute `MessageSize` which encodes this 100.
Please use this and replace the two magic numbers.